### ISSUE TITLE
I donot want use pretrained ，What should I do

### ISSUE THREAD

**Issue by zxjzzz** [ultralytics/ultralytics#9339]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I donot want use pretrained ，What should I do 

like 

model = YOLO('yolov8n.pt')  # load a pretrained model (recommended for training)
# model = YOLO('./fall.yaml').load('yolov8n.pt')  # build from YAML and transfer weights
    
# Train the model
model.train(data='./cat.yaml', epochs=100, imgsz=640)

I donot want yolov8n.pt， because it size is 6m ，

**Comment by glenn-jocher** [comment]
@zxjzzz hello! If you prefer not to use a pretrained model and wish to start training from scratch, you can specify your model architecture with a YAML file without loading the pretrained weights. Here's a small adjustment to your code example:

```python
model = YOLO('./fall.yaml')  # Initialize the model with your YAML configuration

# Train the model
model.train(data='./cat.yaml', epochs=100, imgsz=640)
```

In this way, your model will begin training without utilizing the pretrained weights from `yolov8n.pt`. This approach might take longer to converge to a good accuracy since it's starting from scratch. Best of luck with your training! 😊

**Comment by xiaoshuomin** [comment]
@glenn-jocher hello! I have the same problem with not using a pretrained model. And i have run your code example:
model = YOLO('./fall.yaml') 
model.train(data='./cat.yaml', epochs=100, imgsz=640)
But why the argument pretrained in ./runs/detect/train/args.yaml is always true. Thanks!

**Comment by glenn-jocher** [comment]
@xiaoshuomin hey there! 😊 It looks like the `pretrained` argument in `args.yaml` might default to `True` as a part of the internal configuration setup. If you're starting your training from scratch and don't wish to use pretrained weights, you can explicitly set the `pretrained` flag to `False` in your code:
```python
model = YOLO('./fall.yaml')
model.train(data='./cat.yaml', epochs=100, imgsz=640, pretrained=False)
```
This should ideally override the default behavior and reflect in your `args.yaml` as well. Happy training! 🚀

**Comment by xiaoshuomin** [comment]
@glenn-jocher Thank you for your answer. But, i want to know whether there is start training from scratch if the pretrained is true in ‘model = YOLO('./fall.yaml')’ rather than ‘model = YOLO('./fall.yaml').load('yolov8n.pt')’.

**Comment by glenn-jocher** [comment]
Hey there! 🌟 When you use `model = YOLO('./fall.yaml')`, it initializes a model based on the architecture defined in `'./fall.yaml'`. Setting `pretrained=True` (or leaving it at its default value, which is `True`) will attempt to load pretrained weights if available in the specified path. However, if you don't explicitly load a pretrained model with `.load('yolov8n.pt')`, and no pretrained weights are found or specified, then effectively, your model will start training from scratch!

So, to clarify, using `model = YOLO('./fall.yaml')` with `pretrained=True` but without specifying a `.load(...)` path will only start training from scratch if there are no pretrained weights automatically found. Always happy to help with your YOLOv8 journey! 🚀

**Comment by Dian-Yi** [comment]
> I run these code:
> 
> ```
> yolo_model = YOLO('yolov8n.yaml')
> yolo_model.train(data='scripts/mydata.yaml', epochs=10, workers=1, pretrained=False)
> ```
> 
> but it will download `yolov8n.pt`, i think it use pretrained, so how can i do?
> 
> ```
> Freezing layer 'model.22.dfl.conv.weight'
> AMP: running Automatic Mixed Precision (AMP) checks with YOLOv8n...
> Downloading https://github.com/ultralytics/assets/releases/download/v8.1.0/yolov8n.pt to 'yolov8n.pt'...
> ```

when you use Automatic Mixed Precision (AMP) in train,  In `utils/checks`, 
```
    try:
        from ultralytics import YOLO

        assert amp_allclose(YOLO("yolov8n.pt"), im)
        LOGGER.info(f"{prefix}checks passed ✅")
```
if  'yolov8n.pt' not in you dir , it must download it from url.

**Comment by glenn-jocher** [comment]
Hey there! 🌟 It looks like the framework attempts to download `yolov8n.pt` during the AMP checks, even if you've set `pretrained=False`. This is intended behavior meant for verification purposes and doesn't impact your training starting from scratch. Your training process still adheres to the `pretrained=False` as you specified, so no worries there!

The download happens as a correctness check and doesn't influence the initial weights your model starts with if you've explicitly disabled `pretrained`. Your setup for training without using pretrained weights is correct. Keep going, and happy training! 🚀
